src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -96,13 +128,15 @@ public IndexableRecord convertRecord(SinkRecord record, String index, String typ
     final Object value;
     if (!ignoreSchema) {
       schema = preProcessSchema(record.valueSchema());
-      value = preProcessValue(record.value(), record.valueSchema(), schema);
+      value = record.value() == null ? null : preProcessValue(record.value(), record.valueSchema(), schema);
     } else {
       schema = record.valueSchema();
       value = record.value();
     }
 
-    final String payload = new String(JSON_CONVERTER.fromConnectData(record.topic(), schema, value), StandardCharsets.UTF_8);
+    final String payload = value == null
```

Comment:
nit : formatting looks a little weird for readability. At first look, i read it as payload = value :)

Reply:
Fair enough--will replace the couple of ternary operators with vanilla if/else statements.

Reply:
ternary is fine. May be move it to next line or use parentheses or both

Reply:
Alright, will do.

Reply:
Not sure I agree. Use ternary operators liberally.

Reply:
Also, please don't use unneeded parentheses for readability.

Reply:
+1 for liberal use of ternary and avoiding unnecessary parens.

Reply:
Actually, I'd recommend using `value != null ? ...` rather than `value == null ? ...` to make it more readable and to avoid misreads like @mageshn (and me!).

Reply:
I like the idea of inverting the condition to improve readability, will include that in the next commit. Let me know if readability is still lacking though, definitely not opposed to vanilla if/else if all else (pun intended) fails.